app/models/compute_resource.rb

```diff
@@ -244,7 +244,12 @@ def compute_profile_attributes_for(id)
 
   def vm_compute_attributes_for(uuid)
     vm = find_vm_by_uuid(uuid)
-    vm.attributes.reject{|k,v| k == :id }
+    vm_attrs = vm.attributes.reject{|k,v| k == :id }
```

Comment:
Cloning AWS hosts is currently failing with `undefined method `attributes' for nil:NilClass` , at least if the Host is associated with an EC2 virtual machine that doesn't exist any more.